python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -60,9 +62,41 @@ def get(self, mine_no_or_guid=None):
             # Handle MapView request
             _map = request.args.get('map', None, type=str)
             if _map and _map.lower() == 'true':
-                records = MineMapViewLocation.query.filter(MineMapViewLocation.latitude != None)
-                result = list((map(lambda x: x.json_for_map(), records)))
-                return {'mines': result}
+
+                # Below caches the mine map response object in redis with a timeout.
+                # Generating and jsonifying the map data takes 4-7 seconds with 50,000 points,
+                # so caching seems justified.
+                #
+                # TODO: Use some custom representation of this data vs JSON. The
+                # json string is massive (with 50,000 points: 16mb uncompressed, 2.5mb compressed).
+                # A quick test using delimented data brings this down to ~1mb compressed.
+                map_result = cache.get(MINE_MAP_CACHE)
```

Comment:
Any reason why you're explicitly setting and getting from cache instead of using the header?
`@cache.cached(timeout=43140, key_prefix=MINE_MAP_CACHE)`

My understanding was if it's in cache, flask will just return the json response from the cache. but if not, it'll run the whole function and return data from there?

Reply:
Because it is explicit and readable. As well, since this map=true code path isn't on a separate endpoint I thought this would have lower risk of unintended consequences for the search and other uses. I don't want the search results to get cached or the map data to be returned on a search.

Reply:
Hmm 🤔 Okay that makes sense to be safe. But I dont think anything is using this method, except when you pass map=true.
I believe you can use cache.memoize to ensure it takes into account the parameters passed to the function.

Reply:
The main page mine list and the advanced search call this same method, there is a just a different code path for mine results. At one point I was going to make this a separate endpoint, but with the way the frontend handles the List / Map tab view it would have needed a whoooole bunch of frontend refactoring as well. Out of scope for now.

Reply:
From my understanding you cant use memoize here since this method takes in a reference to self. When memoize caches the function it uses the function and variables as the key. Since the reference to self changes every time you will just run this and cache it every time instead of grabbing the value from the cache.